webdriver/tests/bidi/browsing_context/context_created/original_opener.py

```diff
@@ -0,0 +1,63 @@
+import pytest
+from webdriver.bidi.modules.script import ContextTarget
+
+from .. import assert_browsing_context
+
+pytestmark = pytest.mark.asyncio
+
+CONTEXT_CREATED_EVENT = "browsingContext.contextCreated"
+
+
+@pytest.mark.parametrize("type_hint", ["tab", "window"])
+async def test_original_opener_null(bidi_session, wait_for_event, wait_for_future_safe, subscribe_events, type_hint):
+
+    await subscribe_events([CONTEXT_CREATED_EVENT])
+    on_entry = wait_for_event(CONTEXT_CREATED_EVENT)
+
+    top_level_context = await bidi_session.browsing_context.create(type_hint=type_hint)
+
+    context_info = await wait_for_future_safe(on_entry)
+
+    assert_browsing_context(
+        context_info,
+        context=top_level_context["context"],
+        original_opener=None
```

Comment:
This context is created using WebDriver BiDi and I think the spec does not define what opener should be for those, resulting in the null opener id. @whimboo do you interpret the spec differenty?

Reply:
From Puppeteer perspective, we do not need an opener here since Puppeteer itself opened the context.

Reply:
When using the `browsingContext.create` command with `noopener` to create a new tab/window, it's similar to executing a script like `window.open(..., "noopener")`. In both scenarios, the web page doesn't know about the opener, but privileged code for WebDriver should still be able to return the `opener`. There should be no difference in how the window was opened, and it would be more unusual if we observed different payloads in these cases.

But lets see what @jgraham thinks about.

Reply:
@whimboo so which navigable would Firefox use as an opener in this case?

Reply:
This seems like a spec bug. 

We run https://html.spec.whatwg.org/multipage/nav-history-apis.html#window-open-steps which say:
> Let sourceDocument be the [entry global object](https://html.spec.whatwg.org/multipage/webappapis.html#entry-global-object)'s [associated Document](https://html.spec.whatwg.org/multipage/nav-history-apis.html#concept-document-window).

Following "entry global object"'s definition we end up in https://html.spec.whatwg.org/multipage/webappapis.html#entry-execution-context which assumes that there's at least one item in the [JavaScript execution context stack](https://tc39.es/ecma262/#execution-context-stack). However since WebDriver-BiDi is not in fact calling from JS, that isn't true and we have a spec bug. 

I think this was always wrong, but previously we didn't hit any cases where `sourceDocument` was actually used, so it was more or less invisible.

I wonder if we should just update WebDriver to call https://html.spec.whatwg.org/multipage/document-sequences.html#creating-a-new-top-level-traversable directly. In that case we could set the opener to whatever we want, but the obvious options would either be `null` or the (optional) reference context. I think the reference context makes slightly more sense theoretically, but I don't really know what the use cases require.

Reply:
@whimboo Can you unblock this PR and we can follow up on the spec side.

Reply:
We discussed on Matrix and it looks actually fine.